### DISCUSSION TITLE
Can you specify that a project works with .NET 5 and newer?

### DISCUSSION THREAD

**Discussion by mvtec-bergdolll** [dotnet/runtime#95677] [Q&A] [answered]
Assume you want to write a C# example project that is shipped to a diverse set of customers and we don't know what .NET version is installed on the user systems. I'd like to specify the `TargetFramework` in a way that says, if you have .NET 5 or newer use the newest toolchain available. Essentially I want the users to be able to do `dotnet run` without having to modify the .csproj file. So far I have not found any option that would satisfy this requirement.

**Comment by En3Tho** [comment]
Just specifying ```<TargetFramework>net5.0</TargetFramework>``` should work just as you expect.
The thing here is TargetFramework sets a minimum viable runtime/sdk version
You can both run such project on .Net 5+ runtime and build with .Net 5+ sdk

Which sdk your consumers are using is not really your concern as it's up to them to choose whichever they wamt.

For example I can build your project with any sdk I want, be it .Net 5, 6, 7 or 8 you name it.

Note that .Net 5 is outdated and shouldn't really be considered a good target anymore. Net 6 is a better choice as it is still a supported LTS version.

Or maybe I've misunderstood and you need your customers to run the exe with a latest runtime awailable and not just build it?

**Reply by mvtec-bergdolll** [reply]
@En3Tho that doesn't work the way you describe it. To test it, I have .NET 7 installed on my machine and `<TargetFramework>net6.0</TargetFramework>` configured in the csproj. I get:

```
You must install or update .NET to run this application.

App: <PATH>
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '6.0.0' (x64)
.NET location: /usr/share/dotnet

The following frameworks were found:
  7.0.13 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Learn about framework resolution:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=6.0.0&arch=x64&rid=arch-x64
```

Here is the entire csproj:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <Nullable>enable</Nullable>
  </PropertyGroup>

</Project>
```

Personally I had expected that to just work. This is not the UX I want to offer users trying out our examples.

**Comment by huoyaoyuan** [comment, marked as answer]
Consider to add `<RollForward>Major</RollForward>` to your csproj.

**Reply by mvtec-bergdolll** [reply]
Thank you, that's exactly what I was looking for.
